superset/assets/src/explore/components/controls/DateFilterControl.jsx

```diff
@@ -156,18 +163,24 @@ export default class DateFilterControl extends React.Component {
     this.setTypeCustomRange = this.setTypeCustomRange.bind(this);
     this.setTypeCustomStartEnd = this.setTypeCustomStartEnd.bind(this);
     this.toggleCalendar = this.toggleCalendar.bind(this);
+    this.changeTab = this.changeTab.bind(this);
   }
-  componentDidMount() {
+
+  componentWillMount() {
```

Comment:
I am not sure if we need to do this in `componentWillMount`.
Will we mount/unmount multiple times?
Perhaps do it in the constructor instead if it is only initialization?

Reply:
That's a good suggestion and seems to work fine, thanks